### PR TITLE
reorganize guide

### PR DESCRIPTION
**Pull request by tomerd**
motivation: make guide clearer

changes:
* create introduction section
* create "Building" and move relevant content there
* create "Testing" section and move relevant content there
* create  "Deployment to Public Cloud" section, as placeholder for future content

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,70 @@
...
+
+Another important detail about testing for Linux is the `Tests/LinuxMain.swift` file. For versions prior to Swift 5.1, This file provides SwiftPM an index of all the tests it needs to run on Linux. 
+
+- For version Swift 5.1 and later, this file is no longer required and testing with `--enable-test-discovery` seamlessly discovers the tests. This will be the default behvior in future releases of Swift.
+
+- For versions prior to Swift 5.1, it is critical to keep this file up-to-date as you add more unit tests. To regenerate this file, run `swift test --generate-linuxmain` after adding tests. It is also a good idea to include this command as part of the continuous integration setup. For version Swift 5.1 or later, this 
+
+### Testing for production
+
+- For versions Swift 5.1 and later, always test with `--enable-test-discovery` to avoid forgeting tests on Linux.
```
Comment (weissi) [inline README.md]: I'd add

```
To remind users to add the `--enable-test-discovery` flag if they forgot, please add a file called `Tests/LinuxMain.swift` with the following content:

    fatalError("Please use `swift test --enable-test-discovery` to run the tests instead")

```

Comment (weissi) [reply README.md]: @tomerd / @tanner0101 wdyt? I think that's a massive usability improvement if we just tell the users what they need to do there

Comment (weissi) [reply README.md]: proof that folks are confused: https://discordapp.com/channels/431917998102675485/519613337638797315/680069804333596694

Comment (tomerd) [reply README.md]: good idea @weissi: https://github.com/swift-server/guides/pull/2/commits/243e11524238b03fa1eab98dcc6f087eca62286f

Comment (tanner0101) [reply README.md]: +1

**Pull request merged**